Classes/Services/ElasticSearch/ElasticSearch.php

```diff
@@ -0,0 +1,311 @@
+                            'type' => 'keyword'
+                        ],
+                        'collections' => [
+                            'type' => 'keyword'
+                        ],
+                        'hasFiles' => [
+                            'type' => 'keyword'
+                        ],
+                        'creator' => [
+                            'type' => 'keyword'
+                        ],
+                        'creatorRole' => [
+                            'type' => 'keyword'
+                        ]
+
+                    ]
+                ]
+            ]
+        ];
+
+        if (!$this->client->indices()->exists(['index' => $indexName])) {
+            $this->client->indices()->create($paramsIndex);
+        }
+
+    }
+
+    /**
+     * Adds an document to the index.
+     *
+     * @param Document $document
+     */
+    public function index($document)
+    {
+        $data = json_decode($this->elasticsearchMapper->getElasticsearchJson($document));
+
+        if ($data) {
+
+            $data->state = $document->getState();
+            $data->simpleState = DocumentWorkflow::STATE_TO_SIMPLESTATE_MAPPING[$document->getState()];
+            $data->objectIdentifier = $document->getObjectIdentifier();
+
+            $data->creator = $document->getCreator();
+
+            $data->creatorRole = $this->frontendUserHelper->getUserRole($document->getCreator());
+
+            $data->year = $document->getPublicationYear();
+
+            $notes = $document->getNotes();
+
+            if ($notes && is_array($notes)) {
+                $data->notes = $notes;
+            } else {
+                $data->notes = array();
+            }
+
+            $files = $document->getFile();
+            if ($files->count() > 0) {
+                $data->hasFiles = true;
+            } else {
+                $data->hasFiles = false;
+            }
+
+
+            /** @var @var Mods $mods */
+            $mods = new Mods($document->getXmlData());
+
+            $authors = $mods->getAuthors();
+            $publishers = $mods->getPublishers();
+
+            $data->authorAndPublisher = array_merge($authors, $publishers);
+
+            $data->originalSourceTitle = $mods->getOriginalSourceTitle();
+
+            $this->client->index([
+                'refresh' => 'wait_for',
+                'index' => $this->indexName,
+                'id' => $document->getDocumentIdentifier(),
+                'body' => $data
+            ]);
+
+        }
+
+    }
+
+
+    /**
+     * Deletes a document fron the index
+     *
+     * @param string $identifier
+     */
+    public function delete($identifier)
+    {
+        try {
+
+            $params = [
+                'refresh' => 'wait_for',
+                'index' => $this->indexName,
+                'id' => $identifier
+            ];
+
+            $this->client->delete($params);
+
+        } catch (\Exception $e) {
+            // Nothing to be done.
```

Comment:
[code] This is the perfect opportunity for logging.

Reply:
Logging has been added.